api/Transport.ts

```diff
@@ -1,220 +1,68 @@
-      }
-    : never
-}[keyof StatusCodeMap]
+import { TiFAPITransportMiddleware } from "./Middleware"
+import { EndpointSchemaToMiddleware, TiFAPIRequestBody, TiFHTTPMethod } from "./TransportTypes"
 
 const FETCH_RESPONSE_EMPTY_BODY = ""
 
 const log = logger("tif.api.client")
 
 /**
- * Creates a low-level fetch function to fetch from the TiF API.
- *
- * This lower level fetch handles automatic session tracking and
- * response parsing. The {@link TiFAPI} class provides a set of
- * high-level wrapper functions that utilize this low level
- * fetch function.
- *
- * ```ts
- * const apiFetch = tifAPITransport(
- *   TEST_BASE_URL,
- *   middleware
- * )
- *
- * const ResponseSchema = {
- *   status400: z.object({ b: z.string() }),
- *   status200: z.object({ a: z.number() })
- * }
- *
- * const resp = await apiFetch(
- *   {
- *     method: "POST",
- *     endpoint: "/test",
- *     query: { hello: "world", a: 1 },
- *     body: { a: 1, b: "hello" }
- *   },
- *   ResponseSchema
- * )
- *
- * if (resp.status === 200) {
- *   console.log(resp.data.a) // a is inferred to be a number.
- * }
- * ```
+ * Client implementation of TiFAPIMiddleware that fetches data from a given url.
  *
  * @param baseURL the base url of the backend to use.
- * @param loadAuthBearerToken a function that returns the loaded JWT.
- * @returns a function to make an API call.
+ * @param middleware a function to modify the fetch call.
+ * @returns TiFAPIMiddleware to construct an instance of a TiFAPIImplementation.
  */
-export const tifAPITransport = (baseURL: URL, middleware: TiFAPIMiddleware) => {
-  return async <
-    Method extends TiFHTTPMethod,
-    Schemas extends TiFAPIResponseSchemas
-  >(
-    request: TiFAPIRequest<Method>,
-    responseSchemas: Schemas,
-    signal?: AbortSignal
-  ): Promise<TiFAPIResponse<Schemas>> => {
-    try {
-      const resp = await performRequest(request, middleware, baseURL, signal)
-      const json = await tryResponseBody(resp)
-      const schema = tryResponseSchema(resp.status, responseSchemas, json)
-      return {
-        status: resp.status,
-        data: await tryParseBody(json, resp.status, schema)
-      } as TiFAPIResponse<Schemas>
-    } catch (error) {
-      if (!(error instanceof DOMException) || error.name !== "AbortError") {
-        log.error("Failed to make tif API request.", {
-          error,
-          errorMessage: error.message,
-          request
-        })
+export const tifAPIClient = (baseURL: URL, middleware: TiFAPITransportMiddleware): EndpointSchemaToMiddleware => 
+  (endpointName, { httpRequest: { endpoint, method } }) => 
+    async ({ body, query, params, signal } = {}) => {
+      try {
+        const resp = await performRequest(
+          method, 
+          middleware, 
+          urlConstructor(baseURL, endpoint, query, params), 
+          body,
+          signal
+        )
+        const json = await tryResponseBody(resp)
+        return {
+          status: resp.status as any,
```

Comment:
Nit: We could cast to a more specific type here.

Reply:
Good catch thanks. Fixed